tests/components/hunterdouglas_powerview/test_config_flow.py

```diff
@@ -350,3 +352,79 @@ async def test_form_unsupported_device(
     assert result3["result"].unique_id == "A1B2C3D4E5G6H7"
 
     assert len(mock_setup_entry.mock_calls) == 1
+
+
+@pytest.mark.usefixtures("mock_hunterdouglas_hub")
+@pytest.mark.parametrize("api_version", [1, 2, 3])
+async def test_migrate_entry(
+    hass: HomeAssistant,
+    api_version: int,
+) -> None:
+    """Test migrate to newest version."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={"host": "1.2.3.4"},
+        unique_id=MOCK_MAC,
+        version=1,
+        minor_version=1,
+    )
+
+    assert entry.version == 1
+    assert entry.minor_version == 1
+    assert await entry.async_migrate(hass) is False
+
+    entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert entry.version == 1
+    assert entry.minor_version == 2
+    assert await entry.async_migrate(hass) is True
```

Comment:
We should not call `entry.async_migrate` manually, rather just setup the integration and observe `entry` again to see. I think we can also combine this test with the one down below and check both at the same time.

Ideally we add the entity entries before setting up the integration (like we do in a real life scenario where one boots to the new version) and then setup the intergation. HA will execute the migration function and migrate both the config entry and the entities. And then we should observe the entry and the entity registry to if it migrated to what we expect :)